client/web/src/search/home/QueryExamplesHomepage.tsx

```diff
@@ -208,42 +164,49 @@ export const QueryExamplesHomepage: React.FunctionComponent<QueryExamplesHomepag
                             </>
                         )}
                     </div>
-                    <div className={styles.queryExamplesSectionsColumns}>
-                        {exampleSyntaxColumns.map((column, index) => (
-                            // eslint-disable-next-line react/no-array-index-key
-                            <div key={`column-${index}`}>
-                                {column.map(({ title, queryExamples }) => (
-                                    <QueryExamplesSection
-                                        key={title}
-                                        title={title}
-                                        queryExamples={queryExamples}
-                                        onQueryExampleClick={onQueryExampleClick}
-                                    />
-                                ))}
-                                {!!index && (
-                                    <small className="d-block">
-                                        <Link target="blank" to="/help/code_search/reference/queries">
-                                            Complete query reference{' '}
-                                            <Icon role="img" aria-label="Open in a new tab" svgPath={mdiOpenInNew} />
-                                        </Link>
-                                    </small>
-                                )}
-                            </div>
-                        ))}
-                    </div>
+                    <QueryExamplesLayout queryColumns={exampleSyntaxColumns} onQueryExampleClick={onQueryExampleClick} />
                 </div>
             )}
         </div>
     )
 }
 
-interface QueryExamplesSection {
-    title: string
-    queryExamples: QueryExample[]
+interface QueryExamplesLayout {
+    queryColumns: QueryExamplesSection[][]
+    onQueryExampleClick: (id: string | undefined, query: string, slug: string | undefined) => void
+}
+
+export const QueryExamplesLayout: React.FunctionComponent<QueryExamplesLayout> = ({ queryColumns, onQueryExampleClick }) => (
+    <div className={styles.queryExamplesSectionsColumns}>
+        {queryColumns.map((column, index) => (
+            <div key={`column-${queryColumns[0][0].title}`}>
```

Comment:
Shouldn't the key use something from the current item? Maybe something like
```suggestion
            <div key={`column-${queryColumns[index][0].title}`}>
```

Reply:
Good catch, thanks!